mm2src/mm2_db/src/indexed_db/indexed_cursor.rs

```diff
@@ -166,14 +194,18 @@ impl<'transaction, Table: TableSignature> CursorIter<'transaction, Table> {
 pub enum DbCursorEvent {
     NextItem {
         result_tx: oneshot::Sender<CursorResult<Option<(ItemId, Json)>>>,
+        first_result_only: bool,
     },
 }
 
 pub(crate) async fn cursor_event_loop(mut rx: DbCursorEventRx, mut cursor: CursorDriver) {
     while let Some(event) = rx.next().await {
         match event {
-            DbCursorEvent::NextItem { result_tx } => {
-                result_tx.send(cursor.next().await).ok();
+            DbCursorEvent::NextItem {
+                result_tx,
+                first_result_only,
+            } => {
+                result_tx.send(cursor.next(first_result_only).await).ok();
```

Comment:
Instead of making `NextItem` complicated, what do you think adding a new event called `FirstItem`?